src/settings.cpp

```diff
@@ -374,75 +374,70 @@ static void MakeManyOfMany(char *buf, const char *last, const char *many, uint32
 }
 
 /**
- * Convert a string representation (external) of a setting to the internal rep.
- * @param desc SettingDesc struct that holds all information about the variable
- * @param orig_str input string that will be parsed based on the type of desc
- * @return return the parsed value of the setting
+ * Convert a string representation (external) of an integer-like setting to an integer.
+ * @param str Input string that will be parsed based on the type of desc.
+ * @return The value from the parse string, or the default value of the setting.
  */
-static const void *StringToVal(const SettingDesc *desc, const char *orig_str)
+size_t IntSettingDesc::ParseValue(const char *str) const
```

Comment:
why `size_t` as return value? The doxygen suggests it is an integer. And it is cast to an `int64` / `int32` depending on the user.